### DISCUSSION TITLE
Are `params` CSVs autogenerated or used anywhere apart from user docs?

### DISCUSSION THREAD

**Discussion by shnizzedy** [FCP-INDI/fcp-indi.github.com#249] [Q&A] [answered]
As far as I can tell, the CSV [docs/_sources/_static/params](https://github.com/FCP-INDI/fcp-indi.github.com/tree/source/docs/_sources/_static/params) are only used to generate tables like [![table at https://fcp-indi.github.com/docs/nightly/user/output_config#configuration-without-the-gui](https://user-images.githubusercontent.com/5974438/104608873-6bbf3e80-5650-11eb-9a97-5ca0cc5e7f66.png)](https://fcp-indi.github.com/docs/nightly/user/output_config#configuration-without-the-gui) but I don't think they're autogenerated or used to populate that information anywhere else.

Since those descriptions and potential values either are (or should be) reused between the docs, the GUI, and potentially in places like the YAML comments, helpstrings, docstrings, and C-PAC errror and warning messages, it would be good to single-source-of-truth (SSOT) that information.

Now that [configs are nested](https://github.com/FCP-INDI/C-PAC/issues/1278), I think CSVs aren't the best choice for storing that information.

#### 1. tables with complex merged cell format in `Key` column

We could do tables like [![table at https://shnizzedy.github.io/fcp-indi.github.com/docs/nightly/user/output_config#configuration-without-the-gui](https://user-images.githubusercontent.com/5974438/104639933-f6666480-5675-11eb-9835-01ce3c566903.png)](https://shnizzedy.github.io/fcp-indi.github.com/docs/nightly/user/output_config#configuration-without-the-gui) (and do them programmatically once we have a <span title="single source of truth">SSOT</span> set up for this information), but until we have it set up to be handled programmatically, it'll be tedious to update everything to this format.

The main advantage I see to this format is the clarity of where / how / how deep any given key is nested.

Other ideas:

#### 2. direct users to the comments in the default pipeline until we have the <span title="single source of truth">SSOT</span> set up

[fcp-indi.github.com/docs/{version}/user/pipelines/default](https://shnizzedy.github.io/fcp-indi.github.com/docs/nightly/user/pipelines/default) will include that version's default pipeline with YAML-syntax highlighting.

#### 3. keep the CSV-friendly table format and use JSON-style keys
With keys written like ``pipeline_setup.system_config.on_grid.run`` in the `Key` column

---

Other ideas? Thoughts for now? For the short-, medium-, and long-term?

**Comment by sgiavasis** [comment, marked as answer]
For the short/medium term, let's go with #2 for sure. I've been playing with the idea of somehow making the pipeline config itself as the SSOT in some way.

If the comments in the default pipeline say it, that's what it should say everywhere etc.

**Reply by shnizzedy** [reply]
https://github.com/FCP-INDI/fcp-indi.github.com/pull/248 replaces the `params` tables with excerpts from the default pipeline. We will have to update the line numbers in this repo when things in the default pipeline move.
